src/VisualStudio/Core/Def/Storage/VisualStudioCloudCacheStorageService.cs

```diff
@@ -24,27 +25,33 @@ internal class VisualStudioCloudCacheStorageService : AbstractCloudCachePersiste
         internal class ServiceFactory : IWorkspaceServiceFactory
         {
             private readonly IAsyncServiceProvider _serviceProvider;
+            private readonly IThreadingContext _threadingContext;
 
             [ImportingConstructor]
             [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-            public ServiceFactory(SVsServiceProvider serviceProvider)
-                => _serviceProvider = (IAsyncServiceProvider)serviceProvider;
+            public ServiceFactory(SVsServiceProvider serviceProvider, IThreadingContext threadingContext)
```

Comment:
i'm so confused about what pattern we're supposed to use for importating.  i see some code actually importing IAsyncServiceProvider above, so i don't know why we'd use SVsServiceProvider .

Reply:
These types are originally exported here:
https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform?path=/src/Shell/MefHosting/ComponentModelHostImpl/VsComponentModelHostExporter.cs&version=GBmain&line=23&lineEnd=30&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

The import strategy depends on whether you are using MEF 1 or MEF 2 attributes (MEF 1 is more expressive here).